dashboard/app/models/code_review_comment.rb

```diff
@@ -52,8 +52,10 @@ def self.user_can_review_project?(project_owner, potential_reviewer, project_id,
   def compute_is_from_teacher
     # Only should happen if we hard delete the commenter's account
     # via the delete accounts helper.
-    return false if commenter.nil?
+    return false if commenter.nil? || project_owner.nil?
 
-    self.is_from_teacher = commenter.teacher? ? true : false
+    # This is not ideal, as it only checks that the project owner
+    # is a participant in any section instructed by the commenter.
+    self.is_from_teacher = project_owner.teachers.include?(commenter)
```

Comment:
Could we check if the commenter is associated with this project owner's code review group instead? Or would that just have the same effect?

Reply:
I guess if we assume that comments can only come from a) the project owner, b) ppl in your review group, or c) your instructor/teacher (which is true at the moment), then yeah I think we could say `is_from_teacher = !in_review_group && !from_you`?

Maybe as-is this is slightly less correct, but seems clearer to me and not conditional on any changes we might make to who can review code?